src/js/integrations/debugsymbolicator.ts

```diff
@@ -121,25 +118,47 @@ export class DebugSymbolicator implements Integration {
    * Converts ReactNativeFrames to frames in the Sentry format
    * @param frames ReactNativeFrame[]
    */
-  private _convertReactNativeFramesToSentryFrames(
+  private async _convertReactNativeFramesToSentryFrames(
     frames: ReactNativeFrame[]
-  ): StackFrame[] {
+  ): Promise<StackFrame[]> {
+    let getDevServer: any;
+    try {
+      if (__DEV__) {
+        getDevServer = require("react-native/Libraries/Core/Devtools/getDevServer");
+      }
+    } catch (_oO) {
+      // We can't load devserver URL
+    }
     // Below you will find lines marked with :HACK to prevent showing errors in the sentry ui
     // But since this is a debug only feature: This is Fine (TM)
-    return frames.map(
-      (frame: ReactNativeFrame): StackFrame => {
-        const inApp =
-          (frame.file && !frame.file.includes("node_modules")) ||
-          (!!frame.column && !!frame.lineNumber);
-        return {
-          colno: frame.column,
-          filename: frame.file,
-          function: frame.methodName,
-          in_app: inApp,
-          lineno: inApp ? frame.lineNumber : undefined, // :HACK
-          platform: inApp ? "javascript" : "node" // :HACK
-        };
-      }
+    return Promise.all(
+      frames.map(
+        async (frame: ReactNativeFrame): Promise<StackFrame> => {
+          let inApp = !!frame.column && !!frame.lineNumber;
+          inApp =
+            inApp &&
+            // tslint:disable-next-line: strict-type-predicates
+            frame.file !== undefined &&
+            !frame.file.includes("node_modules") &&
+            !frame.file.includes("native code");
+
+          const newFrame: StackFrame = {
+            colno: frame.column,
+            filename: frame.file,
+            function: frame.methodName,
+            in_app: inApp,
+            lineno: inApp ? frame.lineNumber : undefined, // :HACK
+            platform: inApp ? "javascript" : "node" // :HACK
```

Comment:
Given that we're now filtering out both node modules _and_ native code, would it be better to make the default `undefined`? Or to test for `frame.file.includes("native code")`?